static/js/components/Loading.js

```diff
@@ -72,6 +72,36 @@ export const PostLoading = () => (
   </div>
 )
 
+export const withLoadingRender = ({
+  LoadingComponent,
+  loaded,
+  errored,
+  notAuthorized,
+  notFound,
+  render
+}: LoadingProps & {
+  LoadingComponent: Class<React.Component<*, *>> | Function,
+  render: Function
+}) => {
+  if (notFound) {
+    return <NotFound />
+  }
+
+  if (notAuthorized) {
+    return <NotAuthorized />
+  }
+
+  if (errored) {
+    return <div className="errored">Error loading page</div>
+  }
+
+  if (!loaded) {
+    return <LoadingComponent />
+  }
+
+  return <div className="loaded">{render()}</div>
+}
```

Comment:
this always returns a react component, so I think in `withLoading` below and on `ChannelModerationPage` you could just call this component in JSX instead of as a function call.

`withLoadingRender` is not clearest name I think, maybe something like `LoadingRenderDeferrer` or `DelayedRenderer` or something like that? I'm not sure what is a great name for this, but something that captures more what it's doing would be good I think.